### DISCUSSION TITLE
Custom sound for LocalNotifications

### DISCUSSION THREAD

**Discussion by joshbuker** [ionic-team/capacitor#5201] [Q&A] [answered]
Hi, I can't find any documentation on how to do LocalNotifications with a custom sound.

```javascript
LocalNotifications.schedule({
  notifications: [
    {
      title: options.title,
      body: options.body,
      id: options.id,
      schedule: { at: options.notify_at },
      sound: 'example.wav'
    },
  ],
});
```

^ This will successfully schedule a notification, but I cannot for the life of me change the notification sound to a new file. Where should this file live, and what string should be used for the sound parameter?

For context, I'm using [Quasar](https://quasar.dev) with the capacitor mode for Android.

**Comment by jcesarmobile** [comment, marked as answer]
I think the docs are wrong. They say that the sound property is supported in Android 26+, but as far as I remember, on Android 26+ the way of using sounds is by creating a channel with the sound configured and provide the channel on the notification object. It can't be done with the notification's sound property.

You can configure it globally for all notifications if you add the sound to the plugin configuration:
https://capacitorjs.com/docs/apis/local-notifications#configuration

**Reply by joshbuker** [reply]
I've tried using `capacitor.config.json` and it will change the icon color to green, but does not affect the sound at all (still uses default chime):

```json
{
  "appId": "com.example",
  "appName": "Example App",
  "bundledWebRuntime": false,
  "npmClient": "yarn",
  "webDir": "www",
  "plugins": {
    "LocalNotifications": {
      "smallIcon": "ic_stat_icon_config_sample",
      "iconColor": "#00FF00",
      "sound": "example.wav"
    }
  }
}
```

I've also tried creating a custom channel, with similar lack of results. It either uses the default chime, or has no sound at all. I've never been able to get it to play the custom sound.

**Reply by jcesarmobile** [reply]
The first time you launch the app with the plugin installed a default channel is created, so if you installed the app before setting the sound the channel is already created and setting the sound have no effect, you can't change the sound of an existing channel. 
Try uninstalling the app and installing it again.
If that still doesn't work, would be good if you could provide a sample app, and move information about your testing device (vendor, OS version).

**Reply by joshbuker** [reply]
I'm using an emulator, not a physical device. It's set to `Pixel 5 API 30` atm.

I would assume it re-installs every time I build, but maybe that's an incorrect assumption?

I'm using android studio for debian linux:
![Screenshot from 2021-11-03 11-11-04](https://user-images.githubusercontent.com/1802598/140168197-8661a9d3-3b55-4296-bbb1-d7ea65d9f7ea.png)

**Reply by jcesarmobile** [reply]
It requires a manual uninstall to reset the channels, permissions and other settings, when you re run it overwrites the existing app, but doesn't reset settings.
I've never tried the sound on an emulator, but some things don't work there.

**Reply by joshbuker** [reply]
So I'm not sure if I'm a dumbass, or if it's just the way the default chime works that is really confusing.

Uninstalling and reinstalling got it to go from the default chime to being silent like my other attempts. I then realized there was a volume button for the emulator...I upped the volume and the custom sound is working exactly as expected... I suspect every time it was silent, it was actually playing my custom sound but the emulator was on mute.

So uh, for those in the future, if you can get it to go from the default chime to being silent, make sure your sound is on for the emulator. :man_facepalming: 

Thank you!

**Edit:** Quick clarification for those finding this in the future, you can uninstall in an emulator by going to the apps within settings and uninstalling it like you would on a normal device. Then turn off the emulator and rerun it. Android studio will automatically reinstall when you hit run. I did this by clicking run and then doing the uninstall, but you can also just run the app straight from AVD manager and do it that way.
